### PR TITLE
Setup CI for dlang

### LINKED ISSUE #610: main.yaml needs support for dlang
Right now, running CI doesn't discover a usable toolchain

cc: @yilabs @mw66

### PR DESCRIPTION
**Pull request by adsharma**
Fixes #610

### CODE REVIEW

**Comment by adsharma** [comment]
@mw66: CI run here: https://github.com/py2many/py2many/actions/runs/10016619688

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -162,6 +162,20 @@ jobs:
...
+        with:
+          token: ${{ github.token }}
+
+      - name: Install D Packages
+        if: ${{ !env.FOCUS || env.FOCUS == 'dlang' }}
+        shell: bash
+        run: |
+          dub init -n .
+          dub add dfmt
+          dub fetch dmft --cache=local
```
Comment (adsharma) [inline .github/workflows/main.yml:177]: @mw66: I'm out of depth here. This is failing with:

```
  Error Fetching dmft failed: No package dmft was found matching the dependency >=0.0.0
 ```

**Comment by adsharma** [comment]
So `dfmt` has been fetched, but I'm not sure how to get it into $PATH so the tox can find it. Need help.

**Comment by mw66** [comment]
> So `dfmt` has been fetched, but I'm not sure how to get it into $PATH so tox can find it. Need help.

As last resort: Create a symlink to /usr/bin?

**Comment by jayvdb** [comment]
IMO you would want to change https://github.com/py2many/py2many/blob/main/pyd/__init__.py#L13 so that it is doing `dub run --yes dfmt -- ...` .

**Comment by mw66** [comment]
> IMO you would want to change https://github.com/py2many/py2many/blob/main/pyd/__init__.py#L13 so that it is doing `dub run --yes dfmt -- ...` .

I never tried that, and then it requires`dub` always be in the PATH.

**Comment by adsharma** [comment]
@mw66: There are 9 test failures in this [run](https://github.com/py2many/py2many/actions/runs/10021373709/job/27699961053). Please look into them.

**Comment by mw66** [comment]
> @mw66: There are 9 test failures in this [run](https://github.com/py2many/py2many/actions/runs/10021373709/job/27699961053). Please look into them.

I saw error like this:

AssertionError: '// g[160 chars]) {\n  const L = seq.length;\n  foreach (_; io[475 chars]n}\n' != '// g[160 chars]) {\nconst L = seq.length;\nforeach (_; iota(0[418 chars]\n\n'


It's a dfmt format issue.

How to run this ci test locally?

**Comment by adsharma** [comment]
Trying #617 to see if it fixes things.

**Pull request merged**